### ISSUE TITLE
A sanity check before training the model

### ISSUE THREAD

**Issue by dustinksi** [autonomousvision/sdfstudio#109]
**Is your feature request related to a problem?** 
Yes, the training system does not consider the GPU memory allocation problem before evaluation. This may cause the training process to terminate prematurely during eval due to insufficient memory, resulting in loss of training results.

**Describe the solution you’d like**
The sanity check of the GPU memory before training ensures the memory requirements of both training and evaluation can be met to avoid interruption during training. 

**Describe alternatives you’ve considered**
Refer to the design of PyTorch Lightning to check if the GPU memory usage exceeds the GPU memory limit before training.

**Additional context**
No other context or screenshots

In summary, I think it's better to check the GPU memory requirements of both the training and evaluation process before training process and make appropriate adjustments to ensure that the GPU memory usage of the entire process does not exceed the GPU memory limit to avoid training interruption or loss of results due to insufficient memory. This is a very reasonable requirement. I think this feature would be very helpful in improving the stability and reliability of training.
